### PR TITLE
Enhancement: Add Gradle Executable to Bamboo Dockerfile

### PR DESCRIPTION
**Pull request by ole-ve**

### CODE REVIEW

File: bamboo/Dockerfile [hunk 1]
```diff
@@ -74,6 +74,7 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     software-properties-common=0.9* \
     apt-utils=2.0.* \
     maven=3.6.* \
+    gradle \
```
Comment (Mtze) [inline bamboo/Dockerfile]: Why did you add gradle here?

File: bamboo/Dockerfile [hunk 2]
```diff
@@ -84,5 +85,9 @@ RUN update-alternatives --set java /usr/lib/jvm/java-8-openjdk-arm64/bin/java
 # Add file in /artemis/bin/mvn that uses the correct java version and passes all arguments to mvn
 RUN mkdir /artemis && mkdir /artemis/bin && printf '#!/bin/bash\nJAVA_HOME=/usr/lib/jvm/java-17-oracle /usr/bin/mvn "$@"\n' > /artemis/bin/mvn && chmod 777 /artemis/bin/mvn
 
+# Add the newest gradle version
```
Comment (Hialus) [inline bamboo/Dockerfile]: I think we should lock the Gradle version here.
The versions for some of the other dependencies are also locked and Gradle is known to have some minor incompatibilities between the versions, which I would try to avoid.
I would at least fix the major version (maybe also the minor, as we did it for the other dependencies)

File: bamboo/Dockerfile (hunk 1 above)
Comment (Hialus) [reply bamboo/Dockerfile]: I also think that adding it here as well is not necessary.
Either combine the two additions here or only keep the second one.

**Review by Hialus — CHANGES_REQUESTED** [review]

File: bamboo/Dockerfile (hunk 1 above)
Comment (ole-ve) [reply bamboo/Dockerfile]: Changed in 003d7e9.

File: bamboo/Dockerfile (hunk 2 above)
Comment (ole-ve) [reply bamboo/Dockerfile]: Changed in 003d7e9, version is freezed to 7.3.

**Review by Hialus — APPROVED** [review]
Thanks for applying the changes. 
Looks good to me now. Only one optional comment 😉 
Approve 👍🏻

File: bamboo/Dockerfile [hunk 3]
```diff
@@ -84,5 +84,9 @@ RUN update-alternatives --set java /usr/lib/jvm/java-8-openjdk-arm64/bin/java
 # Add file in /artemis/bin/mvn that uses the correct java version and passes all arguments to mvn
 RUN mkdir /artemis && mkdir /artemis/bin && printf '#!/bin/bash\nJAVA_HOME=/usr/lib/jvm/java-17-oracle /usr/bin/mvn "$@"\n' > /artemis/bin/mvn && chmod 777 /artemis/bin/mvn
 
+# Add gradle version 7.3
+RUN add-apt-repository ppa:cwchien/gradle
+RUN apt-get update
+RUN apt-get install gradle=7.3-0ubuntu1 -y --no-install-recommends
```
Comment (Hialus) [inline bamboo/Dockerfile]: I personally wouldn't lock the minor version as well (I would just do 7.3*)
But this is not strictly necessary.

**Review by Hialus — APPROVED** [review]
Changes look good now.
I haven't tested it locally though.

**Comment by ole-ve** [comment]
> Changes look good now. I haven't tested it locally though.

FYI At least I started the container with the latest command (7.3*) and can confirm that it works.

**Review by Mtze — APPROVED** [review]

**Comment by ole-ve** [comment]
Abandoned: Gradle Wrapper will be used instead which is shipped within the project files and therefore does not need any local installation of Gradle.

**Comment by ole-ve** [comment]
Since making files executable (gradlew in this case) is not possible within packaged JARs like the Artemis Server, the installed Gradle version will be used instead of the Gradle wrapper.

**Comment by Hialus** [comment]
Question:
Why do you even need to execute the gradle wrapper when it is packaged inside of the Artemis Server Jar? The gradle wrapper inside of the ASJ is only used as a template for the repositories.
When the Bamboo Builds are executed the gradle wrapper is not inside the ASJ but in a repository. After checking this repository out the files are on the normal file system and can be made executable if need be.

I also still think that using the gradle wrapper is the more elegant solution.
So this change should still be unnecessary, or am I missing something?

**Comment by ole-ve** [comment]
> Question: Why do you even need to execute the gradle wrapper when it is packaged inside of the Artemis Server Jar? The gradle wrapper inside of the ASJ is only used as a template for the repositories. When the Bamboo Builds are executed the gradle wrapper is not inside the ASJ but in a repository. After checking this repository out the files are on the normal file system and can be made executable if need be.
> 
> I also still think that using the gradle wrapper is the more elegant solution. So this change should still be unnecessary, or am I missing something?

The file itself has not to be executed itself but the permission for this file has to be set to be executable (chmod +x). The files (including gradlew) are copied by reading the content from an input stream of the specific resource and writing this content to the new location. When doing this, file permissions are obviously not kept.
When the build server then tries to execute this file, the corresponding permission is not set and therefore cannot be executed. It may be worth a try to set the permission when executing the build on the build server but I have to check if the Bamboo agent even has these permissions.

**Comment by ole-ve** [comment]
Again, the decision falls in favor to the Gradle wrapper.
